### ISSUE TITLE
Strange NoneType errors with Terms when not accessed directly from an Ontology object

### ISSUE THREAD

**Issue by Melnorme1984** [althonos/pronto#60]
I'm getting some very unusual errors with the latest version of Pronto.

I've got this method for reading the ontology:

```
def read_psimi():
    return pronto.Ontology(DATA_DIRECTORY + PSIMI_FILENAME)
```

If I access a term in the ontology like this:

```
psimi = read_psimi()
psimi_root = psimi[ROOT_DETMETHOD_ID]
```

Then I can use `psimi_root` however I want.

However, if I do this:

`psimi_root = read_psimi()[ROOT_DETMETHOD_ID]`

I get a broken `psimi_root` that gives errors if I try to print it or get its subclasses or superclasses.

`AttributeError: 'NoneType' object has no attribute 'name'`
`AttributeError: 'NoneType' object has no attribute 'get_relationship'`

It seems this problem might be happening whenever the Ontology object is not in the local scope. So even if I do get a functional `psimi_root`, if I return it from a method to a scope where the original Ontology is not available, it becomes broken. Some kind of garbage collection thing?

**Comment by althonos** [comment]
This is expected (there is a small paragraph [here](https://pronto.readthedocs.io/en/stable/api/pronto.Ontology.html#pronto.Ontology.imports) although it should be made clearer I guess) because of how the data is internally managed: the `Term` instances you get let you modify the `Ontology` directly and that is reflected in the other views you get. 

Because of this, your `Ontology` must outlive any terms that are spawned, or you will start getting that `None` error. I could make that error explicit instead so that you know what goes on here.

**Comment by Melnorme1984** [comment]
Yes, I just realized more clearly the scope-related nature of the problem and edited my post. Kind of messed up, isn't it?

**Comment by althonos** [comment]
I'm used to code in Rust, so I see this kind of pattern all the time, but it is true that in Python it's a bit more unusual. As a design choice, it forces the user to store the `Ontology` and treat it preciously, while the terms are cheap to create and can be created and deleted without really paying attention.

> By the way, the term.id field still works when this happens.
At first `Term.id` was also a proxy getter for the internal data, but after experimenting I realised it was the cause of some bad performance because `id` is the only field that is accessed often by the internal code, so I made it an actual field instead (which is fine because you still can't change its value, otherwise you would probably fuck the whole internal referencing up).
